geowrangler/datasets/nightlights.py

```diff
@@ -0,0 +1,526 @@
+# AUTOGENERATED! DO NOT EDIT! File to edit: notebooks/14_datasets_nightlights.ipynb (unless otherwise specified).
+
+__all__ = [
+    "get_eog_access_token",
+    "clear_eog_access_token",
+    "setup_eog_auth_headers",
+    "download_url",
+    "unzip_eog_gzip",
+    "clip_raster",
+    "URLFORM",
+    "EOG_VIIRS_DATA_TYPE",
+    "EOG_PRODUCT",
+    "EOG_PRODUCT_VERSION",
+    "EOG_COVERAGE",
+    "make_url",
+    "make_clip_hash",
+    "generate_clipped_raster",
+    "generate_clipped_metadata",
+    "get_clipped_raster",
+    "generate_nightlights_feature",
+]
+
+
+# Internal Cell
+
+
+import gzip
+import hashlib
+import json
+import os
+import shutil
+from pathlib import Path
+from types import SimpleNamespace
+from urllib.error import HTTPError
+from urllib.parse import urlparse
+
+import numpy as np
+import requests
+from fastcore.net import urldest
+from loguru import logger
+from shapely.geometry import box
+
+import geowrangler.raster_process as rp
+import geowrangler.raster_zonal_stats as rzs
+from .utils import make_report_hook, urlretrieve
+
+# Internal Cell
+DEFAULT_EOG_CREDS_PATH = "~/.eog_creds/eog_access_token"
+EOG_ENV_VAR = "EOG_ACCESS_TOKEN"
+NIGHTLIGHTS_CACHE_DIR = "~/.cache/geowrangler/nightlights"
+
+# Cell
+
+
+# Retrieve access token
+def get_eog_access_token(
+    username,
+    password,
+    save_token=False,
+    save_path=DEFAULT_EOG_CREDS_PATH,
+    set_env=True,
+    env_token_var=EOG_ENV_VAR,
+):
+    params = {
+        "client_id": "eogdata_oidc",
+        "client_secret": "2677ad81-521b-4869-8480-6d05b9e57d48",
+        "username": username,
+        "password": password,
+        "grant_type": "password",
+    }
+    token_url = (
+        "https://eogauth.mines.edu/auth/realms/master/protocol/openid-connect/token"
+    )
+    response = requests.post(token_url, data=params)
+    access_token_dict = json.loads(response.text)
+    access_token = access_token_dict.get("access_token")
+
+    if save_token:
+        logger.info(f"Saving access_token to {save_path}")
```

Comment:
Is there a reason we save this to a file versus generating a new one per request? A common edge case with this is the tokens expiring while we are still downloading it so it becomes less error prone to just a fetch a new one per request.

Reply:
Doesn't look like their access token expires at all. Storing it (read/write only by owner) allows more convenient access after first login so they never have to think about it again.

Reply:
got it. I assumed it was a jwt but on further inspection it isn't.  i am not super comfortable on us handling  saving it to a plaintext file given it is a secret but since the default is save false im okay with this